### PR TITLE
fix: theme not changing when system color scheme changes

### LINKED ISSUE #399: https://bencher.dev/ theme doesn't change when system color scheme changes

### PR DESCRIPTION
**Pull request by hamirmahal**
fixes #399.

### CODE REVIEW

**Comment by hamirmahal** [comment]
# With this change

[Screencast from 05-26-2024 02:03:38 PM.webm](https://github.com/bencherdev/bencher/assets/43425812/3fea58bf-2b3a-48f8-93e8-d53adcb95586)

File: services/console/src/components/navbar/theme/ThemeToggle.astro [hunk 1]
```diff
@@ -60,7 +60,6 @@ import { THEME_TOGGLE_ID, ThemeId } from "./theme";
 
   const initTheme = () => {
     const theme = getTheme();
-    setTheme(theme);
```
Comment (hamirmahal) [inline services/console/src/components/navbar/theme/ThemeToggle.astro:63]: Users who manually toggle the theme will still have their theme preference remain the same, regardless of system theme changes.

File: services/console/src/components/navbar/theme/ThemeToggle.astro [hunk 2]
```diff
@@ -59,8 +59,11 @@ import { THEME_TOGGLE_ID, ThemeId } from "./theme";
   }
 
   const initTheme = () => {
-    const theme = getTheme();
-    setTheme(theme);
+    // Only set the theme if the user has a preference for one.
+    const theme = localStorage.getItem(THEME_KEY);
+    if (theme) {
+      setTheme(theme);
+    }
```
Comment (hamirmahal) [inline services/console/src/components/navbar/theme/ThemeToggle.astro:66]: This lets the website theme automatically update when a user's system theme changes, while still respecting users' theme preferences that they set manually.

**Comment by epompeii** [comment]
Thank you for spotting this and for the PR @hamirmahal !

I have opted to fix this issue by refactoring the code to reuse the same logic for reading from local storage (https://github.com/bencherdev/bencher/commit/00fc5a044dc6ad4f73444451caa3792edbfd4a6a)
Therefore, I'm going to close the PR. I will keep the issue (https://github.com/bencherdev/bencher/issues/399) open though until things get deployed though.

**Comment by hamirmahal** [comment]
You're welcome @epompeii! That works for me.

**Pull request closed without merging**